pkg/options/types.go

```diff
@@ -145,11 +144,11 @@ func (n *NodeType) GetNodeFieldSelector() string {
 		klog.InfoS("Using node type is nil")
 		return EmptyFieldSelector()
 	}
-	pattern := "[^a-zA-Z0-9_,-]+"
-	re := regexp.MustCompile(pattern)
-	result := re.ReplaceAllString(n.String(), "")
-	klog.InfoS("Using node type", "node", result)
-	return fields.OneTermEqualSelector("spec.nodeName", result).String()
+	nodeName := n.String()
+	if nodeName == "" {
```

Comment:
Can you add a comment here, why we're handling this case specially?

Reply:
Added.